R/testing.R

```diff
@@ -0,0 +1,136 @@
+#' Conducting Score Tests for Interaction
+#'
+#' Conduct score tests comparing a fitted model and a more general alternative
+#' model.
+#'
+#' There are two tests available here:
+#'
+#' \bold{Asymptotic Test}
+#'
+#' This is based on the classical variance component test to construct a
+#' testing procedure for the hypothesis about Gaussian process function.
+#'
+#' \bold{Bootstrap Test}
+#'
+#' When it comes to small sample size, we can use bootstrap test instead, which
+#' can give valid tests with moderate sample sizes and requires similar
+#' computational effort to a permutation test.
+#'
+#' @param formula_int A symbolic description of the model with interaction.
+#' @param label_names A character string indicating all the interior variables
+#' included in each predictor.
+#' @param Y Reponses of the dataframe.
+#' @param X1 The first type of factor in the dataframe (could contains several
+#' subfactors).
+#' @param X2 The second type of factor in the dataframe (could contains several
+#' subfactors).
+#' @param Kernlist The kernel library containing several kernels given by user.
+#' @param mode A character string indicating which tuning parameter criteria is
+#' to be used.
+#' @param strategy A character string indicating which ensemble strategy is to
+#' be used.
+#' @param beta A numeric value specifying the parameter when strategy = "exp".
+#' @param test A character string indicating which test is to be used.
+#' @param lambda A numeric string specifying the range of noise to be chosen.
+#' The lower limit of lambda must be above 0.
+#' @param B A numeric value indicating times of resampling when test = "boot".
+#' @return \item{pvalue}{p-value of the test.}
+#' @author Wenying Deng
+#' @seealso method: \code{\link{kernelGenerate}}
+#'
+#' mode: \code{\link{tuning}}
+#'
+#' strategy: \code{\link{ensemble}}
+#' @references Xihong Lin. Variance component testing in generalised linear
+#' models with random effects. June 1997.
+#'
+#' Arnab Maity and Xihong Lin. Powerful tests for detecting a gene effect in
+#' the presence of possible gene-gene interactions using garrote kernel
+#' machines. December 2011.
+#'
+#' Petra Bu ̊zˇkova ́, Thomas Lumley, and Kenneth Rice. Permutation and
+#' parametric bootstrap tests for gene-gene and gene-environment interactions.
+#' January 2011.
+#' @examples
+#'
+#'
+#' ##testing(formula_int = Y ~ X1 * X2,
+#' ##label_names = list(X1 = c("x1", "x2"), X2 = c("x3", "x4")),
+#' ##Y, X1, X2, Kernlist, mode = "loocv", strategy = "erm",
+#' ##beta = 1, test = "boot", lambda = exp(seq(-5, 5)), B = 100)
+#'
+#'
+#' @export testing
+testing <- function(formula_int, label_names, Y, X1, X2, Kernlist,
```

Comment:
similar to ensemble.R, write separate functions `test_asymp` and `test_boot`, then call these methods in main function using `match.arg `+ `do.call`